src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDocumentMappingService.cs

```diff
@@ -498,10 +498,11 @@ public async Task<(Uri MappedDocumentUri, LinePositionSpan MappedRange)> MapToHo
                     {
                         // We're at an edge.
 
-                        if (span.Length > 0 &&
-                            classifiedSpan.AcceptedCharacters == AcceptedCharactersInternal.None)
+                        if (classifiedSpan.SpanKind is SpanKindInternal.MetaCode or SpanKindInternal.Transition)
                         {
-                            // Non-marker spans do not own the edges after it
```

Comment:
This comment was not particularly great at conveying what it was actually looking for. I've written my best understanding of what this condition was actually trying to figure out.